src/core/CheckBox/base.ts

```diff
@@ -0,0 +1,3 @@
+import { CheckboxProps } from '@material-ui/core/Checkbox';
+
+export type BaseCheckboxProps = Omit<CheckboxProps, 'disabled' | 'checked'>;
```

Comment:
Omiting disabled, checked doesn't make sense here....as we are adding it again in props